src/schema-processor/json/parser.ts

```diff
@@ -102,9 +110,25 @@ export class Parser {
       };
     }
 
+    const mz = await credential.merklize(opts.merklizeOpts);
+
+    const credentialType = Parser.findCredentialType(mz);
+
     const subjectId = credential.credentialSubject['id'];
 
-    const slots = this.parseSlots(credential, jsonSchemaBytes);
+    const { slots, nonMerklized } = await this.parseSlots(mz, credential, credentialType);
+
+    // if schema is for non merklized credential, root position must be set to none ('')
+    // otherwise default position for merklized position is index.
+    if (!nonMerklized) {
```

Comment:
maybe put in one if - (!nonMerklized && opts?.merklizedRootPosition === MerklizedRootPosition.None)